packages/next/client/dev/error-overlay/hot-dev-client.js

```diff
@@ -357,14 +364,23 @@ function tryApplyUpdates(onHotUpdateSuccess) {
   }
 
   // https://webpack.js.org/api/hot-module-replacement/#check
-  module.hot.check(/* autoApply */ true).then(
-    (updatedModules) => {
-      handleApplyUpdates(null, updatedModules)
-    },
-    (err) => {
-      handleApplyUpdates(err, null)
-    }
-  )
+  module.hot
+    .check(/* autoApply */ false)
```

Comment:
Disable `autoApply` and apply manually instead, so we may call `onBeforeHotUpdate` right before.